core/modules/replica/IngestServerConnection.cc

```diff
@@ -404,93 +405,105 @@ void IngestServerConnection::_loadDataIntoTable() {
+                    sqlTable,
+                    sqlFullOverlapTable,
+                    sqlDatabase + "." + h.conn->sqlId(ChunkedTable(table, lsst::qserv::DUMMY_CHUNK, not overlap).name()),
+                    sqlDatabase + "." + h.conn->sqlId(ChunkedTable(table, lsst::qserv::DUMMY_CHUNK, overlap).name())
+                };
+                for (auto&& table: tablesToBeCreated) {
+                    statements.push_back(
+                        "CREATE TABLE IF NOT EXISTS " + table + " LIKE " + sqlProtoTable
+                    );
+                    statements.push_back(
+                        "ALTER TABLE " + table + " ADD PARTITION IF NOT EXISTS (PARTITION " + sqlPartition +
+                            " VALUES IN (" + to_string(_transactionId) + "))"
+                    );
+                }
+
+                // An additional step for the current request's table
+                if (table == _table) {
+                    statements.push_back(
+                        "LOAD DATA INFILE " + h.conn->sqlValue(_fileName) +
+                            " INTO TABLE " + (_isOverlap ? sqlFullOverlapTable : sqlTable) +
+                            " FIELDS TERMINATED BY " + h.conn->sqlValue(string() + _columnSeparator)
+                    );
+                }
             }
-            statements.push_back(
-                "LOAD DATA INFILE " + conn->sqlValue(_fileName) +
-                    " INTO TABLE " + (_isOverlap ? sqlFullOverlapTable : sqlTable) +
-                    " FIELDS TERMINATED BY " + conn->sqlValue(string() + _columnSeparator)
-            );
         } else {
 
             // Regular tables are expected to exist in the database before
             // attempting data loading.
 
-            string const sqlTable = sqlDatabase + "." + conn->sqlId(_table);
+            string const sqlTable = sqlDatabase + "." + h.conn->sqlId(_table);
 
             statements.push_back(
                 "ALTER TABLE " + sqlTable + " ADD PARTITION IF NOT EXISTS (PARTITION " + sqlPartition +
                     " VALUES IN (" + to_string(_transactionId) + "))"
             );
             statements.push_back(
-                "LOAD DATA INFILE " + conn->sqlValue(_fileName) +
+                "LOAD DATA INFILE " + h.conn->sqlValue(_fileName) +
                     " INTO TABLE " + sqlTable +
-                    " FIELDS TERMINATED BY " + conn->sqlValue(string() + _columnSeparator)
+                    " FIELDS TERMINATED BY " + h.conn->sqlValue(string() + _columnSeparator)
             );
         }
         for (auto&& statement: statements) {
             LOGS(_log, LOG_LVL_DEBUG, context << __func__ << "  statement: " << statement);
         }
-        conn->execute([&statements](decltype(conn) const& conn_) {
-            conn_->begin();
+        // Note that each statement gets executed in its own transaction to eliminate
+        // a probability of deadlocking on the DML operations creating tables.
```

Comment:
Not sure what this means? "DML operations creating tables" -- creating table is a DDL, and though that it implicitly commits transactions. And MyISAM is not transactional.

Reply:
You're right about the transactions & MyISAM. In this case the proposed modification makes no sense. I just forgot that it's not InnoDB. A reason why I tried this within a transaction was to avoid occasional deadlocks (have seen them a coupe of times when the Ingest servers were under heavy load) on sequences of the table `CREATE`, `ALTER` (to add a MySQL partition) and `LOAD DATA INFILE` statements. It turns out table locking is trickier if using MySQL partitions. So, let me leave this code as it's now. I will investigate this issue as separate effort.

Reply:
At the mean time I'm eliminating the confusing comment.